File: Modules/Payments/Models/Payment.php
<?php

namespace Modules\Payments\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Modules\Payments\Enums\PaymentStatus;
use Modules\Sessions\Models\Session;
use Modules\User\Models\User;

class Payment extends Model
{
    protected $fillable = [
        'user_id',
        'session_id',
        'amount',
        'currency',
        'status',
        'gateway',
        'gateway_transaction_id',
        'gateway_response',
        'paid_at',
    ];

    protected $casts = [
        'status' => PaymentStatus::class,
        'gateway_response' => 'array',
        'paid_at' => 'datetime',
        'amount' => 'decimal:2',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function session(): BelongsTo
    {
        return $this->belongsTo(Session::class);
    }
}

File: Solid.Api/Database/Entities/Payment.cs
namespace Solid.Api.Database.Entities;

public sealed class Payment
{
    public long Id { get; set; }

    public long UserId { get; set; }

    public long? SessionId { get; set; }

    public decimal Amount { get; set; }

    public string Currency { get; set; } = "EGP";

    public string Status { get; set; } = string.Empty;

    public string Gateway { get; set; } = string.Empty;

    public string? GatewayTransactionId { get; set; }

    public string? GatewayResponse { get; set; }

    public DateTime? PaidAt { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public User User { get; set; } = null!;

    public TherapySession? Session { get; set; }
}
